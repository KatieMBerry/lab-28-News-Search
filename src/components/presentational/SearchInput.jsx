import React from 'react';
import PropTypes from 'prop-types';

const SearchInput = ({ searchInput, onChange }) => (
    <>
        <label htmlFor="searchInput">Search</label>
        <input
            id="searchInput"
            type="text"
            value={searchInput}
            onChange={onChange} />
    </>
);

SearchInput.propTypes = {
    searchInput: PropTypes.string,
    onChange: PropTypes.func.isRequired
};

export default SearchInput;
