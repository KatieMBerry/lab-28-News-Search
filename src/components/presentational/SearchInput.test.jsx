import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SearchInput from './SearchInput';

describe('SearchInput component', () => {
    afterEach(() => cleanup());
    it('renders SearchInput', () => {
        const { asFragment } = render(<SearchInput
            searchInput='my_search_query'
            onChange={jest.fn()}
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
