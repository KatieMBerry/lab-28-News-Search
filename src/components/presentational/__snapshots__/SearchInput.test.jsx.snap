// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders SearchInput 1`] = `
<DocumentFragment>
  <label
    for="searchInput"
  >
    Search
  </label>
  <input
    id="searchInput"
    type="text"
    value="my_search_query"
  />
</DocumentFragment>
`;
